[BUG] Build fails on FreeBSD due to `-Wl,-rpath,${ESPEAK_NG_RPATH_ORIGIN}`

### Important Instructions ⚠️

Before submitting this issue, please **read carefully** and confirm the following:

- [x] I am using the **latest version** of `sherpa-onnx`. If not, I will upgrade and check if the issue persists.
- [x] I understand that **issues not reproducible on the latest version** may be closed without response.
- [x] I have verified that the error is reproducible using **only `sherpa-onnx`**, not code from my own project.
- [x] I will provide **detailed steps** to reproduce the issue.
- [x] I will provide **complete logs** and error messages.
- [x] I understand that **issues not following these instructions** may be closed or receive no response (如果直接删掉这个issue template 里的步骤，我们很有可能**不会回复**您的issue).

---

### Describe the issue

A clear and concise description of the problem.

### Steps to reproduce

On FreeBSD 16.x:
1. git clone https://github.com/k2-fsa/sherpa-onnx
2. mkdir build && cd build
3. cmake -GNinja ..
4. ninja

### Expected behavior

sherpa-onnx builds

### Environment
- `sherpa-onnx` version: 20ac56ef404efbb4300effc9b22c8ca038313566
- OS: FreeBSD 16.0-CURRENT #39 wipbsd-n287383-e78048c091b6
- Python version: Python 3.11.15
- onnxruntime version: onnxruntime-1.26.0 (from FreeBSD ports)
- GPU (if applicable): N/A
- NPU (if applicable): N/A

### Logs

Build fails with:
```
FAILED: [code=1] bin/sherpa-onnx-offline-tts-play 
: && /usr/bin/c++ -O3 -DNDEBUG  sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts-play.dir/sherpa-onnx-offline-tts-play.cc.o sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts-play.dir/microphone.cc.o -o bin/sherpa-onnx-offline-tts-play  -Wl,-rpath,"\$ORIGIN:/usr/local/lib:"  lib/libsherpa-onnx-portaudio_static.a  lib/libsherpa-onnx-core.a  -Wl,-rpath,/../lib  -Wl,-rpath,/../../../sherpa_onnx/lib  -ljack  /usr/local/lib/libasound.so  lib/libkaldi-native-fbank-core.a  lib/libkissfft-float.a  lib/libkaldi-decoder-core.a  lib/libsherpa-onnx-kaldifst-core.a  lib/libssentencepiece_core.a  -lpthread  /usr/local/lib/libonnxruntime.so  -lm  lib/libsherpa-onnx-fstfar.a  lib/libsherpa-onnx-fst.a  lib/libpiper_phonemize.a  lib/libespeak-ng.a  -lm  -Wl,-rpath,  lib/libucd.a && :
/usr/local/bin/x86_64-unknown-freebsd16.0-ld: lib/libespeak-ng.a(common.c.o): in function `towlower2':
common.c:(.text+0x6bb): undefined reference to `ucd_tolower'
/usr/local/bin/x86_64-unknown-freebsd16.0-ld: lib/libespeak-ng.a(readclause.c.o): in function `clause_type_from_codepoint':
readclause.c:(.text+0x9): undefined reference to `ucd_lookup_category'
/usr/local/bin/x86_64-unknown-freebsd16.0-ld: readclause.c:(.text+0x12): undefined reference to `ucd_properties'
/usr/local/bin/x86_64-unknown-freebsd16.0-ld: lib/libespeak-ng.a(ssml.c.o): in function `attrnumber':
ssml.c:(.text+0x19a7): undefined reference to `ucd_tolower'
/usr/local/bin/x86_64-unknown-freebsd16.0-ld: lib/libespeak-ng.a(translate.c.o): in function `TranslateClauseWithTerminator':
translate.c:(.text+0x22fc): undefined reference to `ucd_toupper'
/usr/local/bin/x86_64-unknown-freebsd16.0-ld: translate.c:(.text+0x2313): undefined reference to `ucd_toupper'
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```

Fixed by editing _deps/espeak_ng-src/src/libespeak-ng/CMakeLists.txt to remove
```
target_link_libraries(espeak-ng PRIVATE "-Wl,-rpath,${ESPEAK_NG_RPATH_ORIGIN}")
```

Thanks for sharing the complete logs!

```
[524/652] : && /usr/bin/c++ -O3 -DNDEBUG  sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts.dir/sherpa-onnx-offline-tts.cc.o -o bin/sherpa-onnx-offline-tts  -Wl,-rpath,"\$ORIGI
N:/usr/local/lib:"  lib/libsherpa-onnx-core.a  -Wl,-rpath,/../lib  -Wl,-rpath,/../../../sherpa_onnx/lib  lib/libkaldi-native-fbank-core.a  lib/libkissfft-float.a  lib/libkaldi-de
coder-core.a  lib/libsherpa-onnx-kaldifst-core.a  lib/libssentencepiece_core.a  -lpthread  /usr/local/lib/libonnxruntime.so  -lm  lib/libsherpa-onnx-fstfar.a  lib/libsherpa-onnx-
fst.a  lib/libpiper_phonemize.a  lib/libespeak-ng.a  -lm  -Wl,-rpath,  lib/libucd.a && :
FAILED: [code=1] bin/sherpa-onnx-offline-tts
: && /usr/bin/c++ -O3 -DNDEBUG  sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts.dir/sherpa-onnx-offline-tts.cc.o -o bin/sherpa-onnx-offline-tts  -Wl,-rpath,"\$ORIGIN:/usr/loc
al/lib:"  lib/libsherpa-onnx-core.a  -Wl,-rpath,/../lib  -Wl,-rpath,/../../../sherpa_onnx/lib  lib/libkaldi-native-fbank-core.a  lib/libkissfft-float.a  lib/libkaldi-decoder-core
.a  lib/libsherpa-onnx-kaldifst-core.a  lib/libssentencepiece_core.a  -lpthread  /usr/local/lib/libonnxruntime.so  -lm  lib/libsherpa-onnx-fstfar.a  lib/libsherpa-onnx-fst.a  lib
/libpiper_phonemize.a  lib/libespeak-ng.a  -lm  -Wl,-rpath,  lib/libucd.a && :
/usr/local/bin/x86_64-unknown-freebsd16.0-ld: lib/libespeak-ng.a(common.c.o): in function `towlower2':
common.c:(.text+0x6bb): undefined reference to `ucd_tolower'
```

In the above log,

```
-lm  -Wl,-rpath,  lib/libucd.a 
```

this part does not look right.  There should be some path after `-rpath,`

---

Inside `build/_deps/espeak_ng-src/CMakeLists.txt`, we have
```

if(NOT APPLE)
  set(ESPEAK_NG_RPATH_ORIGIN "$ORIGIN")
else()
  set(ESPEAK_NG_RPATH_ORIGIN "@loader_path")
endif()
```

and in `build/_deps/espeak_ng-src/src/libespeak-ng/CMakeLists.txt`, we have
```
target_link_libraries(espeak-ng PRIVATE "-Wl,-rpath,${ESPEAK_NG_RPATH_ORIGIN}")
```

`${ESPEAK_NG_RPATH_ORIGIN}` should not be empty in your case.

Can you print its value during cmake configuration?

--- a/cmake/espeak-ng-for-piper.cmake
+++ b/cmake/espeak-ng-for-piper.cmake
@@ -1,9 +1,8 @@
 function(download_espeak_ng_for_piper)
   include(FetchContent)
 
-  set(espeak_ng_URL  "https://github.com/csukuangfj/espeak-ng/archive/f6fed6c58b5e0998b8e68c6610125e2d07d595a7.zip")
-  set(espeak_ng_URL2 "https://hf-mirror.com/csukuangfj/sherpa-onnx-cmake-deps/resolve/main/espeak-ng-f6fed6c58b5e0998b8e68c6610125e2d07d595a7.zip")
-  set(espeak_ng_HASH "SHA256=70cbf4050e7a014aae19140b05e57249da4720f56128459fbe3a93beaf971ae6")
+  set(espeak_ng_URL  "https://github.com/csukuangfj/espeak-ng/archive/ed530aa113046142eb5115cf2fc9157854d0ffe1.zip")
+  set(espeak_ng_HASH "SHA256=e4e262cbe34f7fe21f91f1ba3397f2728e1f30eafbae7853f2b753a9ed13f0dd")
 
   set(BUILD_ESPEAK_NG_TESTS OFF CACHE BOOL "" FORCE)
   set(USE_ASYNC OFF CACHE BOOL "" FORCE)
@@ -21,11 +20,11 @@ function(download_espeak_ng_for_piper)
   # If you don't have access to the Internet,
   # please pre-download kaldi-decoder
   set(possible_file_locations
-    $ENV{HOME}/Downloads/espeak-ng-f6fed6c58b5e0998b8e68c6610125e2d07d595a7.zip
-    ${CMAKE_SOURCE_DIR}/espeak-ng-f6fed6c58b5e0998b8e68c6610125e2d07d595a7.zip
-    ${CMAKE_BINARY_DIR}/espeak-ng-f6fed6c58b5e0998b8e68c6610125e2d07d595a7.zip
-    /tmp/espeak-ng-f6fed6c58b5e0998b8e68c6610125e2d07d595a7.zip
-    /star-fj/fangjun/download/github/espeak-ng-f6fed6c58b5e0998b8e68c6610125e2d07d595a7.zip
+    $ENV{HOME}/Downloads/espeak-ng-ed530aa113046142eb5115cf2fc9157854d0ffe1.zip
+    ${CMAKE_SOURCE_DIR}/espeak-ng-ed530aa113046142eb5115cf2fc9157854d0ffe1.zip
+    ${CMAKE_BINARY_DIR}/espeak-ng-ed530aa113046142eb5115cf2fc9157854d0ffe1.zip
+    /tmp/espeak-ng-ed530aa113046142eb5115cf2fc9157854d0ffe1.zip
+    /star-fj/fangjun/download/github/espeak-ng-ed530aa113046142eb5115cf2fc9157854d0ffe1.zip
   )
 
   foreach(f IN LISTS possible_file_locations)
@@ -33,7 +32,6 @@ function(download_espeak_ng_for_piper)
       set(espeak_ng_URL  "${f}")
       file(TO_CMAKE_PATH "${espeak_ng_URL}" espeak_ng_URL)
       message(STATUS "Found local downloaded espeak-ng: ${espeak_ng_URL}")
-      set(espeak_ng_URL2 )
       break()
     endif()
   endforeach()
@@ -41,7 +39,6 @@ function(download_espeak_ng_for_piper)
   FetchContent_Declare(espeak_ng
     URL
       ${espeak_ng_URL}
-      ${espeak_ng_URL2}
     URL_HASH          ${espeak_ng_HASH}
   )
 

